src/server/pps/server/api_server_test.go

```diff
@@ -405,3 +408,52 @@ func TestSetClusterDefaults(t *testing.T) {
 		require.True(t, defaults.CreatePipelineRequest.Autoscaling, "default autoscaling should be true after SetClusterDefaults")
 	})
 }
+
+func TestCreatePipelineV2(t *testing.T) {
+	ctx := pctx.TestContext(t)
+	env := realenv.NewRealEnv(ctx, t, dockertestenv.NewTestDBConfig(t))
+
+	_, err := env.PPSServer.SetClusterDefaults(ctx, &pps.SetClusterDefaultsRequest{
+		ClusterDefaultsJson: `{"create_pipeline_request": {"datum_tries": 17, "autoscaling": true}}`,
+	})
+	require.NoError(t, err, "SetClusterDefaults failed")
+
+	repo := "input"
+	pipeline := "pipeline"
+	require.NoError(t, env.PachClient.CreateRepo(pfs.DefaultProjectName, repo))
+	var pipelineTemplate = `{
```

Comment:
Is this a space vs. tabs thing?  I would indent the nested JSON as though it were filling out a go struct.  Alternatively, I'd also consider declaring this a `map[string]any`, marshaling, and using that as the template.  I think you could actually skip templating in this case and it could be a Go function instead, however.

Reply:
> I would indent the nested JSON as though it were filling out a go struct.

👍🏻 

>  I think you could actually skip templating in this case and it could be a Go function instead, however.

I could, but I think that using a template to generate the string is more understandable.